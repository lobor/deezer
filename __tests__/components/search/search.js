import React from 'react';
import Search from 'components/search';
import renderer from 'react-test-renderer';

describe('test search button', () => {
  it('test render', () => {
    const component = renderer.create(
      <Search />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('check submit', () => {
    let submit = jest.fn();
    const component = renderer.create(
      <Search submit={submit} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onSubmit()
    expect(submit).toBeCalled();
  })
})
