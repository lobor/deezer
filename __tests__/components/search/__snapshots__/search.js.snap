// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search button check submit 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <input
    className="input"
    name="search"
    placeholder="Search"
    type="text"
  />
  <button
    className="btn btn-primary"
  >
    ⌕
  </button>
</form>
`;

exports[`test search button test render 1`] = `
<form
  className="search"
  onSubmit={undefined}
>
  <input
    className="input"
    name="search"
    placeholder="Search"
    type="text"
  />
  <button
    className="btn btn-primary"
  >
    ⌕
  </button>
</form>
`;
